### ISSUE TITLE
Better default behaviour for CLI reads-search command

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#702]
The client command line interface provides an easy to use interface to the server functions. For example, we can use 
```
$ python client_dev.py datasets-search http://localhost:8000 -O json
{"description": null, "name": "1kg-p3-subset", "id": "MWtnLXAzLXN1YnNldA=="}
```
to run the datasets/search command. To make this as user-friendly as possible we try to do the obvious thing when arguments are not provided. For example, when we run the variants-search without arguments like
```
$ python client_dev.py variants-search http://localhost:8000
```
we find all ``VariantSets`` on the server, and run the variants-search command on each one. This makes quick exploration of the data on the server much faster, as we don't need to manually copy IDs to get a quick look at the data.

However, the ``reads-search`` command does not have this useful feature.

Edit ``ga4gh/cli.py`` and find SearchReadsRunner. 

- Defined a method ``getAllReadGroups`` that returns all ReadGroups on the server using ``getAllRedGroups``.
- Then, in SearchReadsRunner.run(), If ``readGroupIds`` has not been specified, iterate over all readGroups and search for all reads in each readGroup. See SearchVariantsRunner for an example of how this can be done.
- While processing a given readGroup, if the referenceId has been specified, use this reference. If not, find the ReferenceSet that the readGroup refers to, and iterate over all references in this referenceSet.

The [client API docs](http://ga4gh-reference-implementation.readthedocs.org/en/develop/api.html) will be useful for this.

**Comment by pkrusche** [comment]
For the last part, the ReadGroup objects only seem to have a referenceSetId, what we'd need is a referenceId. I am guessing this referenceId will come from each read, so this would need to be done in the backend?

**Comment by dcolligan** [comment]
Fixed in #738
